byteps/_keras/callbacks.py

```diff
@@ -54,13 +67,17 @@ def _average_metrics_in_place(self, logs):
         # Reduce every metric among workers. Sort metrics by name
         # to ensure consistent order.
         for metric, value in sorted(logs.items()):
-            if metric not in self.variables:
-                self.variables[metric], self.push_pull_ops[metric] = \
-                    self._make_variable(metric, value)
+            if bps._executing_eagerly():
+                reduced_logs[metric] = \
+                    bps.allreduce(self.backend.constant(value, name=metric)).numpy()
```

Comment:
I don't think we have allreduce api. Should it be `bps.push_pull`?

Reply:
right, it should be push_pull